### ISSUE TITLE
[Bug]: Vehicle mass setting doesn't always work

### ISSUE THREAD

**Issue by BigboiKilljoy** [Yimura/YimMenu#2862]
### Describe the bug

There's a setting for vehicle mass which should allow a small normally light vehicle, to hit like the heaviest thing in the game. instead it's just hitting like the same weak pathetic thing it would hit like without mods

### Steps To Reproduce

1. Go to vehicle handling
2. Set vehicle mass to 50K
3. Ram something, and clearly the vehicles mass (99% of the time) is unchanged. 

### Expected Behavior

Set vehicle mass to 50K 
Ram an NPC car and it explodes 

### Logs and/or screenshots

```
<paste your logs here>
```

### Store

Steam

### Language

English

### YimMenu Language

English (US)

### Additional context

```json
{
    "m_gravity": 11.531999588012695,
    "m_handling_data": {
        "m_acceleration": 0.9599999785423279,
        "m_anti_rollbar_bias_front": 1.2000000476837158,
        "m_anti_rollbar_bias_rear": 0.7999999523162842,
        "m_anti_rollbar_force": 0.8999999761581421,
        "m_brake_bias_front": 0.7599999904632568,
        "m_brake_bias_rear": 1.2400000095367432,
        "m_brake_force": 1.430999994277954,
        "m_buoyancy": 85.0,
        "m_camber_stiffness": 0.0,
        "m_centre_of_mass": {
            "x": 0.0,
            "y": 0.0,
            "z": 0.0
        },
        "m_collision_damage_mult": 0.699999988079071,
        "m_curve_lateral": 0.3839724361896515,
        "m_curve_lateral_ratio": 2.604353666305542,
        "m_downforce_multiplier": 0.0,
        "m_downshift": 3.819999933242798,
        "m_drive_bias_front": 0.0,
        "m_drive_bias_rear": 1.1764706373214722,
        "m_drive_inertia": 1.0,
        "m_drive_max_flat_velocity": 51.66667175292969,
        "m_handbrake_force": 0.6000000238418579,
        "m_inertia_mult": {
            "x": 1.0,
            "y": 1.600000023841858,
            "z": 1.600000023841858
        },
        "m_initial_drag_coeff": 0.0010600000387057662,
        "m_initial_drive_force": 0.3619999885559082,
        "m_initial_drive_gears": 9,
        "m_initial_drive_max_flat_vel": 43.05555725097656,
        "m_low_speed_traction_loss_mult": 1.187000036239624,
        "m_mass": 50000.0,
        "m_roll_centre_height_front": 0.09000000357627869,
        "m_roll_centre_height_rear": 0.09000000357627869,
        "m_steering_lock": 0.6981317400932312,
        "m_steering_lock_ratio": 1.4323945045471191,
        "m_suspension_bias_front": 1.0,
        "m_suspension_bias_rear": 1.0,
        "m_suspension_comp_damp": 0.11000000685453415,
        "m_suspension_force": 1.899999976158142,
        "m_suspension_lower_limit": -0.10999999940395355,
        "m_suspension_raise": 0.014999999664723873,
        "m_suspension_rebound_damp": 0.14000000059604645,
        "m_suspension_upper_limit": 0.10000000149011612,
        "m_traction_bias_front": 0.968999981880188,
        "m_traction_bias_rear": 1.031000018119812,
        "m_traction_curve_lateral": 0.42553192377090454,
        "m_traction_curve_max": 2.375,
        "m_traction_curve_min": 2.2209999561309814,
        "m_traction_curve_ratio": 6.666673183441162,
        "m_traction_loss_mult": 1.0,
        "m_traction_spring_delta_max": 0.15000000596046448,
        "m_traction_spring_delta_max_ratio": 6.666666507720947,
        "m_upshift": 3.819999933242798
    }
```
The vehicle mass is correctly saved as 50K 

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by Yimura** [comment]
The mass slider requires a reload of something or should the class data needs to be modified before the first vehicle is spawned. Basically according to FiveM it requires a reload of the client before it works.

**Comment by BigboiKilljoy** [comment]
> The mass slider requires a reload of something or should the class data needs to be modified before the first vehicle is spawned. Basically according to FiveM it requires a reload of the client before it works.

That's all confusing.

**Comment by BigboiKilljoy** [comment]
Can you write down the steps to making the vehicle mass work correctly?

**Comment by Yimura** [comment]
> Can you write down the steps to making the vehicle mass work correctly?

There are no steps, you need to somehow reload the physics of the game so it accepts the new mass of the vehicle. I haven't investigated it any further than finding out from FiveM that it requires a reload of your game (aka. restarting your game basically). And because YimMenu does the modification only after you've loaded the vehicle in it's already too late, you need to find out a method to hijack the physics loading of a vehicle and modify it there before the game can apply the normal mass from the config file.

Basically this requires more fundamental changes to get it working on the fly and combined with the fact that the handling system is one of the oldest parts of YimMenu and hasn't been changed in all this time makes it a royal pain in the ass to make any modifications to it.

**Comment by BigboiKilljoy** [comment]
> > Can you write down the steps to making the vehicle mass work correctly?
> 
> There are no steps, you need to somehow reload the physics of the game so it accepts the new mass of the vehicle. I haven't investigated it any further than finding out from FiveM that it requires a reload of your game (aka. restarting your game basically). And because YimMenu does the modification only after you've loaded the vehicle in it's already too late, you need to find out a method to hijack the physics loading of a vehicle and modify it there before the game can apply the normal mass from the config file.
> 
> Basically this requires more fundamental changes to get it working on the fly and combined with the fact that the handling system is one of the oldest parts of YimMenu and hasn't been changed in all this time makes it a royal pain in the ass to make any modifications to it.

I know that the vehicle mass works sometimes for what ever stupid reason, so i guess at some point ill just fuck around a bunch and pay attention to everything i do. Maybe ill find some steps to making this work

**Comment by BigboiKilljoy** [comment]
Thanks for sharing what you know :)

**Comment by Friuuu** [comment]
yeah like said before it needs reload of something. i have gotten it working in missions where the same car is present like in the first mission modify mass and next mission it will be changed
